src/mechanics/elements/ElementFem.h

```diff
@@ -12,49 +12,71 @@ namespace NuTo
 class ElementFem : public ElementInterface
 {
 public:
-    ElementFem(std::vector<NuTo::NodeSimple*> nodes, const InterpolationSimple& interpolation)
-        : mNodes(nodes)
-        , mInterpolation(interpolation)
+    ElementFem(std::initializer_list<std::reference_wrapper<NuTo::NodeSimple>> nodes,
+               const InterpolationSimple& interpolation)
+        : mInterpolation(interpolation)
     {
+        for (NuTo::NodeSimple& node : nodes)
+            mNodes.push_back(&node);
+        assert(mNodes.size() == interpolation.GetNumNodes());
+        assert(mNodes.front()->GetNumValues() == interpolation.GetDofDimension());
     }
 
     virtual NodeValues ExtractNodeValues() const override
     {
-        int dim = mNodes[0]->GetNumValues();
-        Eigen::VectorXd nodeValues(mNodes.size() * dim);
-        for (size_t i = 0; i < mNodes.size(); ++i)
+        const int dim = GetDofDimension();
+        Eigen::VectorXd nodeValues(GetNumNodes() * dim);
+        for (size_t i = 0; i < GetNumNodes(); ++i)
             nodeValues.segment(dim * i, dim) = mNodes[i]->GetValues();
         return nodeValues;
     }
 
     NMatrix GetNMatrix(NaturalCoords ipCoords) const override
     {
-        return Matrix::N(mInterpolation.GetShapeFunctions(ipCoords), mInterpolation.GetNumNodes(),
-                         mInterpolation.GetDofDimension());
+        return Matrix::N(Interpolation().GetShapeFunctions(ipCoords), Interpolation().GetNumNodes(),
+                         Interpolation().GetDofDimension());
     }
 
-    Eigen::VectorXd GetShapeFunctions(Eigen::VectorXd ipCoords) const override
+    ShapeFunctions GetShapeFunctions(NaturalCoords ipCoords) const override
     {
-        return mInterpolation.GetShapeFunctions(ipCoords);
+        return Interpolation().GetShapeFunctions(ipCoords);
```

Comment:
What is the advantage of using the additional method Interpolation() instead of mInterpolation?

Reply:
I changed `const&` to `reference_wrapper`. The reference wrapper syntax involves a `.get()`, which I found even uglier. Maybe easier to change later. Maybe only preference.